scripts/testramfs/testramfs.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// testramfs tests things, badly
+package main
+
+import (
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"syscall"
+
+	flag "github.com/spf13/pflag"
+
+	"github.com/u-root/u-root/pkg/cpio"
+	"github.com/u-root/u-root/pkg/pty"
+)
+
+const cloneFlags = syscall.CLONE_NEWNS |
+	syscall.CLONE_NEWIPC |
+	syscall.CLONE_NEWNET |
+	syscall.CLONE_NEWPID |
+	syscall.CLONE_NEWUTS |
+	syscall.CLONE_NEWPID
+	//| syscall.CLONE_NEWUSER
+
+var (
+	unshared = flag.Bool("unshared", false, "whether this instance has an unshared name space")
+	noremove = flag.BoolP("noremove", "n", false, "remove tempdir when done")
+	interactive = flag.BoolP("interactive", "i", false, "interactive mode")
+)
+
+func main() {
+	flag.Parse()
+
+	if flag.NArg() != 1 {
+		log.Fatalln("usage: %s <cpio-path>", os.Args[0])
+	}
+
+	c := flag.Args()[0]
+
+	f, err := os.Open(c)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// So, what's the plan here?
+	//
+	// - new mount namespace
+	//   - root mount is a tmpfs mount filled with the archive.
+	//
+	// - new PID namespace
+	//   - archive/init actually runs as PID 1.
+
+	tempDir, err := ioutil.TempDir("", "u-root")
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Don't do a RemoveAll. This should be empty and
+	// an error can tell us we got something wrong.
+	if !*noremove {
```

Comment:
yea yea closure no param needed but I've been watching that kind of thing screw people lately so i'm gradually losing my love for it. The parameter ensures no surprises.